test/mibs.jl

```diff
@@ -1,155 +1,63 @@
-    x = lower_variables[1]
-    @test lower_objective ≈ MOI.ScalarAffineFunction{Float64}([MOI.ScalarAffineTerm(-1.0, x)],0.0)
-    @test lower_sense == MOI.MIN_SENSE
+function runtests()
+    for name in names(@__MODULE__, all = true)
+        if startswith("$(name)", "test_")
+            @testset "$(name)" begin
+                getfield(@__MODULE__, name)()
+            end
+        end
+    end
 end
 
-
-@testset "basic example (3) - integer in upper level" begin
+function test_basic_example_1()
     model = BilevelModel()
-
-    @variable(Upper(model), y)
+    @variable(Upper(model), y, Int)
     @variable(Upper(model), z, Int)
-    @variable(Lower(model), x)
-
+    @variable(Lower(model), x, Int)
     @objective(Upper(model), Min, 3x + y + z)
     @constraints(Upper(model), begin
         u1, x <= 5
         u2, y <= 8
         u3, y >= 0
-        u4, z >=0
+        u4, z >= 0
     end)
-
     @objective(Lower(model), Min, -x)
     @constraint(Lower(model), l1,  x +  y <= 8)
     @constraint(Lower(model), l2, 4x +  y >= 8)
     @constraint(Lower(model), l3, 2x +  y <= 13)
     @constraint(Lower(model), l4, 2x - 7y <= 0)
-
     new_model, lower_variables, lower_objective, lower_constraints, lower_sense = BilevelJuMP._build_single_model(model)
-
-    #Check over number of variables
-    @test length(lower_variables) == 1 # lower
-    @test MOI.get(new_model, MOI.NumberOfVariables()) == (2 + 1) # upper + lower
-
-    #Check over number of constraints
-    @test length(lower_constraints) == (4 + 0) #lower constraint + number of integer variables
-    
-    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.LessThan{Float64}}()) == (2 + 3) # lessthan
-    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}}()) == (2 + 1) # lessthan
-    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.SingleVariable, MOI.Integer}()) == (1 + 0) # number of constraints for integer represantation of the variables
-
-    #Check over the objective function of the lower level
+    @test length(lower_variables) == 1
+    @test MOI.get(new_model, MOI.NumberOfVariables()) == (2 + 1)
+    @test length(lower_constraints) == (4 + 0)
+    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.LessThan{Float64}}()) == (2 + 3)
+    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64},MOI.GreaterThan{Float64}}()) == (2 + 1)
+    @test MOI.get(new_model, MOI.NumberOfConstraints{MOI.SingleVariable, MOI.Integer}()) == 3
     x = lower_variables[1]
     @test lower_objective ≈ MOI.ScalarAffineFunction{Float64}([MOI.ScalarAffineTerm(-1.0, x)],0.0)
     @test lower_sense == MOI.MIN_SENSE
+    solution = BilevelJuMP.solve_with_MibS(model, silent = false)
+    @test solution.status == true
+    @test solution.objective ≈ 8
+    @test solution.nonzero_upper == Dict(0 => 8)
+    @test solution.nonzero_lower == Dict{Int,Float64}()
```

Comment:
For now we just return this solution struct. @joaquimg can handle mapping the solutions back into JuMP-space. That might be a little complicated.